### PR TITLE
use sbt-pekko-build

### PR DESCRIPTION
**Pull request by pjfanning**
allow us to share some build code across projects

### CODE REVIEW

**Comment by mdedetrich** [comment]
@pjfanning I just pushed 2 commits on this branch. The first one just updates from latest `main`, the second one uses the latest sbt-pekko-build and sets the critical sbt `SettingKey` (i.e. `pekkoCoreProject`), it also removes the existing `PekkoInlinePlugin` since this is now being handled by `sbt-pekko-build`

**Comment by mdedetrich** [comment]
Let me check the flags are smart with and without this change

**Review by mdedetrich — APPROVED** [review]
All good from my end (aside from a different ordering of the inline scalac flags everythings the same)

File: build.sbt [hunk 1]
```diff
@@ -26,6 +26,11 @@ commands := commands.value.filterNot { command =>
 
 ThisBuild / reproducibleBuildsCheckResolver := Resolver.ApacheMavenStagingRepo
 
+ThisBuild / pekkoCoreProject := true
```
Comment (He-Pin) [inline build.sbt:29]: add a comment about this `pekkoCoreProject`?

Comment (mdedetrich) [reply build.sbt:29]: I mean it's kind of implied, this repo is the core pekko project?

**Comment by mdedetrich** [comment]
@pjfanning Do you want to cherry pick or should I?

**Comment by pjfanning** [comment]
> @pjfanning Do you want to cherry pick or should I?

You can do this if you like

**Pull request merged**